src/tools/storage-key-recipe-resolver.ts

```diff
@@ -172,3 +182,13 @@ export function findLongRunningArcId(recipe: Recipe): string | null {
   }
   return null;
 }
+
+
+/** Determines if create handle is volatile. */
+function isVolatileCreateHandle(handle: Handle): boolean {
+  if (handle.capabilities.isEmpty()) return true;
+  if (!handle.capabilities.isPersistent &&
+    !handle.capabilities.isTiedToRuntime &&
+    !handle.capabilities.isTiedToArc) return true;
```

Comment:
this seems wrong, TiedToArc does map to volatile right now

Reply:
Given your above comment, I'm going to s/isTiedToArc/isQueryable/, does that sound right to you?